packages/klighd-core/src/search-bar/searchbar-panel.tsx

```diff
@@ -0,0 +1,707 @@
+        this.selectedIndex = -1
+        /* Debug: */ console.log("[SearchBar] setting results:", results)
+    }
+
+    /**
+     * Updates the variable textRes.
+     * @param res : an array containing the text field from each search result match
+     */
+    public setTextRes(res: string[]): void {
+        this.textRes = res
+    }
+
+    @inject(TYPES.IActionDispatcher) protected readonly actionDispatcher: IActionDispatcher
+
+    readonly id: 'search-bar-panel'
+    readonly title: 'Search'
+    readonly position: number = 0
+
+    onUpdate(callback: () => void): void {
+        this.updateCallbacks.push(callback)
+    }
+
+    update(): void {
+        for (const callback of this.updateCallbacks) {
+            callback()
+        }
+    }
+
+    /**
+     * Toggles the visibility of the tag input field
+     */
+    private toggleTagInput(): void {
+        this.tagInputVisible = !this.tagInputVisible
+
+        if (!this.tagInputVisible) {
+            const textInput = document.getElementById('search') as HTMLInputElement 
+            const tagInput = document.getElementById('tag-search') as HTMLInputElement
+            if (tagInput) tagInput.value = ''
+            if (textInput) {
+                textInput.value === '' ? this.resetUI() : this.performSearch()
+            }
+        }
+
+        this.update()
+
+        if (this.tagInputVisible) {
+            setTimeout(() => {
+                const tagInput = document.getElementById('tag-search') as HTMLInputElement
+                if (tagInput) {
+                    tagInput.focus()
+                }
+            }, 0)
+        }
+    }
+
+    /**
+     * Performs the search with both main and tag inputs
+     */
+    private performSearch(): void {
+        const mainInput = document.getElementById('search') as HTMLInputElement
+        const tagInput = document.getElementById('tag-search') as HTMLInputElement
+        
+        const query = mainInput ? mainInput.value : ''
+        const tagQuery = tagInput ? tagInput.value : ''
+
+        this.clearError()
+        
+        this.searched = true
+        const start = performance.now()
+
+        this.actionDispatcher.dispatch(ClearHighlightsAction.create())
+        this.actionDispatcher.dispatch(SearchAction.create(this, SearchBar.ID, query, tagQuery ))
+
+        const end = performance.now()
+        const total = end - start
+        console.log(`[SearchBar] search took ${total.toFixed(10)} ms`)
+        
+        document.addEventListener('keydown', this.handleKeyNavigation)
+    }
+
+    /**
+     * Renders the search bar.
+     * @param vis : should the search bar currently be visible
+     * @param panel : the search bar panel
+     * @returns the search bar panel if vis=true, else an empty div
+     */
+    render(vis: boolean, panel: SearchBarPanel): VNode {
+        if (!vis) {
+            return <div style={{ display: 'none' }}></div>
+        }
+
+        return h('div', {
```

Comment:
please replace all this `h` function with JSX code and hide the styling behind classes outsourced in a .css file.